src/main/java/org/folio/circulation/resources/CheckOutByBarcodeResource.java

```diff
@@ -208,33 +163,69 @@ private String urlForLoan(String id) {
     return String.format("/circulation/loans/%s", id);
   }
 
-  private Result<LoanAndRelatedRecords> addProxyUser(Result<LoanAndRelatedRecords> loanResult,
-    Result<User> getUserResult) {
+  private CompletableFuture<Result<User>> getUserByBarcode(String barcode,
+    UserRepository userRepository, CirculationErrorHandler errorHandler) {
 
-    return loanResult.combine(getUserResult, LoanAndRelatedRecords::withProxyingUser);
+    return userRepository.getUserByBarcode(barcode)
+      .thenApply(r -> errorHandler.handleValidationResult(r, FAILED_TO_FETCH_USER, succeeded(null)));
   }
 
   private Result<LoanAndRelatedRecords> addUser(Result<LoanAndRelatedRecords> loanResult,
-    Result<User> getUserResult) {
+    Result<User> getUserResult, CirculationErrorHandler errorHandler) {
+
+    if (getUserResult.value() == null || errorHandler.hasAny(FAILED_TO_FETCH_USER)) {
```

Comment:
Can this method and previous one be combined into one? This way we can avoid these redundant checks.